### PR TITLE
GEPA: add evaluation adapter seam

### PR DESCRIPTION
**Pull request by XiaoConstantine**
## Summary
- add a GEPA evaluation adapter that snapshots a stable batch once and returns structured per-example evaluation records
- route population evaluation through the adapter so concurrent candidate scoring no longer iterates the shared dataset directly
- add regression coverage for adapter-based candidate evaluation and one-time batch materialization

## Context
- Follow-up parity slice for #226
- This keeps the existing evolutionary loop intact, but creates the internal evaluation seam needed for a future DSPy-style reflective minibatch loop

## Testing
- go test ./...
- golangci-lint run ./...

### CODE REVIEW

File: pkg/optimizers/gepa_evaluation_adapter.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+func (g *GEPA) materializeEvaluationBatch(dataset core.Dataset) []core.Example {
+	if dataset == nil {
+		return nil
+	}
+
+	// Preserve existing GEPA semantics: a non-positive batch size means
+	// "materialize the full dataset" instead of truncating the batch.
+	limit := g.config.EvaluationBatchSize
```
[the remark this answers is not in the document]
Comment (XiaoConstantine) [reply pkg/optimizers/gepa_evaluation_adapter.go]: Fixed in c326ac5.

You were right that the adapter changed the non-positive `EvaluationBatchSize` edge case. `NewGEPA()` normalizes that path for normal callers, but this PR is meant to be a mechanical refactor and we already construct bare `GEPA{}` instances inside the package tests. I restored the pre-adapter behavior for non-positive sizes and added a regression test covering `0` and negative values so the refactor stays behaviorally equivalent on that edge too.
